client/data/media/with-delete-media.js

```diff
@@ -0,0 +1,37 @@
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch, useSelector } from 'react-redux';
+import { useDeleteMediaMutation } from 'calypso/data/media/use-delete-media-mutation';
+import { deleteMedia as deleteMediaAction } from 'calypso/state/media/actions';
+import { gutenframeUpdateImageBlocks } from 'calypso/state/media/thunks';
+import { errorNotice, removeNotice } from 'calypso/state/notices/actions';
+import { requestMediaStorage } from 'calypso/state/sites/media-storage/actions';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+export const withDeleteMedia = createHigherOrderComponent(
+	( Wrapped ) => ( props ) => {
+		const dispatch = useDispatch();
+		const translate = useTranslate();
+		const siteId = useSelector( getSelectedSiteId );
```

Comment:
I wonder if there will be an instance where we'll need to do this for a site different than the selected site. Maybe it's just overkill.

Reply:
Like in a cross-site media library? Hm… 🤔 – the previously passed in `siteId` was derived from `getSelectedSiteId` as well. If you think it's safer we can also make `siteId` a param of the `deleteMedia` mutate function (as it used to be) instead of the hook.

Reply:
I think what you did is cleaner 👍  Just wanted to see what you think.

Reply:
My opinion is that `siteId` and `mediaId` are just two parts of one larger ID, and should always travel together and be passed around using the same convention. Either:
```js
{ mutate } = useDeleteMutation( siteId, mediaId );
mutate();
```
or
```js
{ mutate } = useDeleteMutation();
mutate( { siteId, mediaId } );
```
but _not_ like the current mixed convention:
```js
{ mutate } = useDeleteMutation( siteId );
mutate( { mediaId } );
```

Reply:
> My opinion is that `siteId` and `mediaId` are just two parts of one larger ID, and should always travel together […]

I'm totally fine with your suggestion but wanted to clarify whether that is specific to media in your opinion or would that go for other combinations of such IDs as well? Looking at other mutation functions (like c[lient/data/users/use-update-user-mutation.js](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/users/use-update-user-mutation.js)) we split there as well.

Reply:
Thanks for mentioning the `useUpdateUserMutation` hook. After looking at both, I'd say yes, both of them should use the
```js
{ mutate } = useDeleteMutation();
mutate( { siteId, mediaId } );
```
convention. The main reason is that when the hook is wrapped in a HOC, it's more clear where `siteId` is coming from. The `withUpdateUser` HOC assumes `siteId` is a prop of the outer component. The `withDeleteMedia` HOC reads `siteId` from Redux. We've been choosing rather randomly depending on what the use case was at the moment.

Reply:
Thanks for elaborating! I think that's a good step into developing more formalized conventions for our React Query usage 👍🏻 I'll add a commit which includes changes accordingly.

Reply:
I think Jarda's rationale makes sense here 👍

Reply:
Added https://github.com/Automattic/wp-calypso/pull/58573/commits/7f713922aa753b060d4de9ee38d2763878035ccd